mono/metadata/cominterop.c

```diff
@@ -3664,3 +3664,54 @@ ves_icall_System_Runtime_InteropServices_Marshal_FreeBSTR (gpointer ptr)
 {
 	mono_free_bstr (ptr);
 }
+
+void*
+mono_cominterop_get_com_interface (MonoObject* object, MonoClass* ic)
+{
+#ifndef DISABLE_COM
+	MonoError error;
+
+	if (!object)
+		return NULL;
+
+	if (cominterop_object_is_rcw (object)) {
+		MonoClass *klass = NULL;
+		MonoRealProxy* real_proxy = NULL;
+		if (!object)
+			return NULL;
+		klass = mono_object_class (object);
+		if (!mono_class_is_transparent_proxy (klass)) {
+			g_assert_not_reached ();
+			return NULL;
+		}
+
+		real_proxy = ((MonoTransparentProxy*)object)->rp;
+		if (!real_proxy) {
+			g_assert_not_reached ();
+			return NULL;
+		}
+
+		klass = mono_object_class (real_proxy);
+		if (klass != mono_class_get_interop_proxy_class ()) {
+			g_assert_not_reached ();
```

Comment:
Could you flesh out some of these assertion cases with something that propagates an exception, sets an error, or at least gives an explanatory message in the case of a failure? Assertions like this turn a local bug in a potentially non-critical path into something that can take down an entire application.